calculator_test.go

```diff
@@ -2,23 +2,416 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+// Note Fatal|FatalF are also useful methods,
+// to stop further execution within the same test.
+
+// Express test-case inputs and desired outputs
+type testCase struct {
+	a           float64
+	b           float64
+	want        float64
+	description string
+	errExpected bool
+}
+
+// This repeats the above struct,
+// but b is a slice-of-float64 instead.
+// Re-using the same struct for variadic and non-variadic would be nice,
+// but separate structs makes test more readable.
+type variadicTestCase struct {
+	a           float64
+	b           []float64
+	want        float64
+	description string
+	errExpected bool
+}
+
+// Test case for expressions (string parsed into numbers and an operator)
+type expressionTestCase struct {
+	e           string
+	want        float64
+	description string
+	errExpected bool
+}
+
+// Return a slice of N random numbers from 0-500
+func randFloat64Slice(n int) []float64 {
+	rand.Seed(time.Now().UnixNano())
+	// TODO: Should this be pre-allocated instead of using append()?
```

Comment:
Yes; we know in advance how long `r` will be, so we can allocate all the memory in one operation by calling `make`, instead of having it repeatedly and unpredictably copy and re-allocate every time the slice needs to grow dynamically.